cpp/csp/adapters/websockets/ClientHeaderUpdateAdapter.cpp

```diff
@@ -0,0 +1,26 @@
+#include <csp/adapters/websockets/ClientHeaderUpdateAdapter.h>
+
+namespace csp::adapters::websockets {
+
+ClientHeaderUpdateAdapter::ClientHeaderUpdateAdapter(
+    Engine * engine,
+    Dictionary& properties
+) : OutputAdapter( engine ), m_properties(properties)
+{
+
+};
+
+void ClientHeaderUpdateAdapter::executeImpl()
+{
+    std::vector<std::string> value = input() -> lastValueTyped<std::vector<std::string>>();
+    DictionaryPtr headers = m_properties.get<DictionaryPtr>("headers");
+    // this is hacky but it works...
```

Comment:
if you want it to be less hacky ( with slightly more work ) you can define a csp.Struct class with the key : str and value : str fields, and make headers an array of that.
You would then have to add autogen to generate the c++ headers to access to the data ( or, you dont, and you can look up the fields dynamically too )